### PR TITLE
Use Node's URL API to resolve

### PR DESCRIPTION
**Pull request by alshakero**
I tried moving to Axios instead of Web API XHR. But this meant I should migrate XHR tests to Moxios.
And this in turn meant moving to Mocha + Node (CLI) tests. And Web URL doesn't work in Node. So I had to shim it using Node's URL before moving to Axios.

The shim is loaded only in Node env, to actually test it, you'll have to use the bundle because Webpack assumes a node env when bundling.

To do this, polyserve and go to:

<pre>
http://127.0.0.1:8000/components/Palindrom/test/<b>SpecRunnerBundle</b>.html
</pre>

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
   "devDependencies": {
     "polymer": "*",
     "palindrom-bunny": "Palindrom/palindrom-bunny",
-    "dom-bind-notifier": "*"
+    "dom-bind-notifier": "*",
+    "fast-json-patch": ">=1.1.0"
```
Comment (warpech) [inline bower.json]: Again, can't we use `fast-json-patch` from `node_modules`?

Also `dom-bind-notifer` should be removed while we're at it.

File: src/URL.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/* URL DOM API shim */
+var resolve = require("url").resolve;
+var parse = require("url").parse;
```
Comment (warpech) [inline src/URL.js]: Is `url` a built-in module? Shouldn't it be added to `package.json`?

File: bower.json (hunk 1 above)
Comment (alshakero) [inline bower.json]: Great and cleaner idea, done.

File: src/URL.js (hunk 2 above)
Comment (alshakero) [inline src/URL.js]: Amazing catch, forget the `--save` flag.

**Pull request merged**